my_sprites.py

```diff
@@ -58,13 +66,23 @@ def __init__(
     def max_hp(self):
         return self._max_hp
 
+    @property
+    def hp(self):
+        return self._max_hp
+
+    @hp.setter
+    def hp(self, new_hp):
+        self._hp = max(0, min(new_hp, self.max_hp))
+
```

Comment:
Let's add a getter and a setter for the state. So, rename the state variable to _self._state_ internally.